### PR TITLE
Integrating UniRep into bio_embeddings

### PR DESCRIPTION
**Pull request by konstin**
## PR Description

I'm working on [bio_embeddings](https://gitlab.lrz.de/sacdallago/bio_embeddings) ([poster](https://f1000research.com/posters/9-876)), a tool that embeds sequences using various methods, and we'd like to include UniRep. During integrating this library with bio_embeddings, I came across three minor problems.

Currently, `pip install https://github.com/ElArkk/jax-unirep` and PEP 517 calls in general fail because `from jax_unirep import __version__` in `setup.py` import `jax_unirep/__init__.py`, which requires the dependencies to be present. I've add a workaround in the first commit that avoids importing jax_unirep (see also https://stackoverflow.com/questions/2058802/how-can-i-get-the-version-defined-in-setup-py-setuptools-in-my-package for more details).

The logger setup in `evotuning.py` creates a `evotuning.log` in the current working directory, even if logging has been configured differently elsewhere and evotuning isn't even used. I'd like to move that logging to the correct main-function or script if possible, however unfortunately I could not find a fitting one. It would be great if you could tell me where to move the logger setup.

We'd like to support python 3.6 with bio_embeddings since this is the version on google colab. Since I didn't see any python 3.7 specific code, I've changed the requirement in setup.py.

Aside from these small issues, the library was a pleasure to work with!

### General
1. [x] I have made the PR off a new branch from my fork 
   (`<your_username>`:`<feature-branch_name>`), *not* 
   `<your_username>`:`master`.
2. [x] I have added my changes to the `CHANGELOG.md` file at the top.
3. [x] I have made any necessary changes to the documentation in the `README`. _not applicable_

### Code checks
1. [x] If there are new features implemented, add suitable tests
   in the `tests` directory. _not applicable_
2. [x] If any new dependencies are introduced through the new features,
   add the packages, pinned to a version, to `environment.yml`. _not applicable_
3. [x] Run `make test` in a console in the top level directory
   to make sure all the tests pass.
4. [ ] Run `make format` in a console in the top level directory
   to make the code comply with the formatting standards.

    I tried to run `make format`, but it failed because isort 5 didn't like the options used in the Makefile. With isort 4 and black 19.10b0  I still got a bunch of unrelated changes in files I didn't touch.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
-from setuptools import find_packages, setup
-from jax_unirep import __version__
+from setuptools import setup
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
+with open("jax_unirep/version.py") as fp:
+    version = fp.read().split('"')[1]
```
Comment (ericmjl) [inline setup.py:7]: Nice! I didn't know about this practice beforehand. Thanks for helping us with this!

File: jax_unirep/evotuning.py [hunk 2]
```diff
@@ -29,14 +29,7 @@
...
 )
 
-# setup logger
 logger = logging.getLogger("evotuning")
-logger.setLevel(logging.INFO)
-fh = logging.FileHandler("evotuning.log")
-fh.setLevel(logging.INFO)
-formatter = logging.Formatter("%(asctime)s :: %(levelname)s :: %(message)s")
-fh.setFormatter(formatter)
-logger.addHandler(fh)
```
Comment (ericmjl) [inline jax_unirep/evotuning.py:39]: I've got a proposal for this block of code. What if we hid it inside a function block, and called it when executing evotuning? Proposal:

```python
def setup_evotuning_log():
    # code block gets moved inside here.
    # returns nothing
````

Then, in `fit`, right at the top:

```python
def fit(...):
    """Docstring goes here."""
    setup_evotuning_log()
```

Comment (konstin) [reply jax_unirep/evotuning.py:39]: That would work great! I've updated the code to your proposal.

**Comment by konstin** [comment]
Thank you for the quick review! The poster has already been presented as [recorded talk](https://www.youtube.com/watch?v=NucUA0QiOe0) at ISMB by Christian Dallago, the project lead.

**Comment by ericmjl** [comment]
Everything looks good to me. @ElArkk I know you're on vacation, so I'll merge this first for now, and then cut a release shortly.

**Pull request merged**